docs/coil.md

```diff
@@ -0,0 +1,154 @@
+# Coil for Jetpack Compose
+
+[![Maven Central](https://img.shields.io/maven-central/v/com.google.accompanist/accompanist-coil)](https://search.maven.org/search?q=g:com.google.accompanist)
+
+This library brings easy-to-use [Painter][painter] which can fetch and display images from external sources, such as network, using the [Coil][coil] image loading library.
+
+<img src="https://coil-kt.github.io/coil/logo.svg" width="480" alt="Coil logo">
+
+!!! info
+    If you're migrating from Accompanist 0.7.x or before, please read the [migration](./migration-coilimage) documentation after reading this document.
+
+## `rememberCoilPainter()`
+
+The primary API is via the [`rememberCoilPainter()`][rememberpainter] function. The simplest usage is like so:
+
+```kotlin 
+import androidx.compose.foundation.Image
+import com.google.accompanist.coil.rememberCoilPainter
+
+Image(
+    painter = rememberCoilPainter("https://picsum.photos/300/300"),
+    contentDescription = "My content description",
+)
+```
+
+This painter loads the data passed in, using [Coil][coil], and then draws the resulting image.
+
+You can also customize the Coil [`ImageRequest`](https://coil-kt.github.io/coil/image_requests/) through the `requestBuilder` parameter. This allows usage of things like (but not limited to) transformations:
+
+```kotlin
+import androidx.compose.foundation.Image
+import com.google.accompanist.coil.rememberCoilPainter
+
+Image(
+    painter = rememberCoilPainter(
+        request = "https://picsum.photos/300/300",
+        requestBuilder = {
+            transformations(CircleCropTransformation())
+        },
+    ),
+    contentDescription = "My content description",
+)
+```
+
+## Fade-in animation
+
+This library has built-in support for animating loaded images in, using a [fade-in animation](https://material.io/archive/guidelines/patterns/loading-images.html).
+
+![](crossfade.gif)
+
+
+A `fadeIn: Boolean` parameter is available on [`rememberCoilPainter()`][rememberpainter] (default: `false`). When enabled, a default fade-in animation will be used when the image is successfully loaded:
+
+``` kotlin
+import androidx.compose.foundation.Image
+import com.google.accompanist.coil.rememberCoilPainter
+
+Image(
+    painter = rememberCoilPainter(
+        "https://picsum.photos/300/300",
+        fadeIn = true
+    ),
+    contentDescription = "My content description",
+)
+```
+
+## Custom content
+
+Some times you may wish to display some alternative content whilst the image is loading, or an error has occurred. The painter returned from `rememberCoilPainter()` is an instance of [`LoadPainter`][loadpainter], which is stateful and allows you to display different content as required:
+
+
+``` kotlin
+val painter = rememberCoilPainter("https://picsum.photos/300/300")
+
+Box {
+    Image(
+        painter = painter,
+        contentDescription = "My content description",
+    )
+
+    if (painter.loadState == ImageLoadState.Loading) {
```

Comment:
Use a `when` here to show the other states (with ellided bodys)?

Reply:
Done